pkg/services/sqlstore/user.go

```diff
@@ -146,6 +146,9 @@ func verifyExistingOrg(sess *DBSession, orgId int64) error {
 
 func (ss *SQLStore) getOrCreateOrg(sess *DBSession, orgName string) (int64, error) {
 	var org org.Org
+	org.Created = time.Now()
+	org.Updated = org.Created
+
 	if ss.Cfg.AutoAssignOrg {
 		has, err := sess.Where("id=?", ss.Cfg.AutoAssignOrgId).Get(&org)
```

Comment:
I'm thinking there has to be a `sess.Where("name=?", orgName).Get(&org)`  before these  to respect a GetOrCreate.

What happens after is what's ruled by the `ss.Cfg.AutoAssignOrg`. 

@sakjur can you sanity check this?

Reply:
I agree, this method looks superweird, I don't see that it'd ever actually get an existing org apart from the main org?

Reply:
What I'm not sure about is whether the functionality or the naming is wrong here?

Reply:
I think `sess.Where("name=?", orgName).Get(&org)` should come after the `else` on line 173 - so if `AutoAssignOrg` is set to false, we either create an org with `orgName` or fetch an existing org with `orgName` for this user.

My understanding of the logic is as follows:
* if `AutoAssignOrg == true` - assign user to org with `AutoAssignOrgId` (I don't know why we have the special case of orgId = 1 where we create it if it doesn't exist, that's weird);
* if `AutoAssignOrg == false` - assign user to their own org that's named `orgName`

Reply:
I'm merging this PR. The proper refactor will need to be done at a later stage. That being said, I created an issue to keep track of the needed improvement.

https://github.com/grafana/grafana/issues/66131